docker-compose.yml

```
@@ -30,14 +30,24 @@ services:
       - model_inference_api
 
   frontend:
-      image: basic_image_streamlit
-      build: crypto_sentiment_demo_app/frontend
-      command: streamlit run crypto_sentiment_demo_app/frontend/streamlit_app.py
-      ports:
-        - "8501:8501"
-      volumes:
-        - ./:/root
-      depends_on:
-        - model_inference_api
-      links:
-        - model_inference_api
+    image: basic_image_streamlit
+    build: crypto_sentiment_demo_app/frontend
+    command: streamlit run crypto_sentiment_demo_app/frontend/streamlit_app.py
+    ports:
+      - "8501:8501"
+    volumes:
+      - ./:/root
+    depends_on:
+      - model_inference_api
```

Думаю стоит добавить зависимость от data_provider и прописать healthcheck, чтобы фронт не обращался к data_provider, пока он не запущен

```suggestion
    depends_on:
      model_inference_api:
        condition: service_healthy
      data_provider: 
        condition: service_healthy
```